src/randoop/FailureSet.java

```diff
@@ -6,35 +6,55 @@
 import java.util.Set;
 
 import randoop.operation.MethodCall;
-import randoop.operation.Operation;
 import randoop.sequence.ExecutableSequence;
 import randoop.sequence.MutableSequence;
 import randoop.sequence.MutableVariable;
+import randoop.sequence.Statement;
 import randoop.sequence.Variable;
 
+/**
+ * FailureSet represents a set of failures that occurred in the execution of a particular
+ * {@link ExecutableSequence}. 
+ *
+ */
 public class FailureSet {
   
   private Set<Failure> failures = new LinkedHashSet<Failure>();
   
+  /**
+   * FailureSet.Failure represents an execution failure and holds the
+   * statement and class where the failure occurred.
+   *
+   */
   public static class Failure {
-    public final Operation st;
-    public final Class<?> viocls;
-    public Failure(Operation st, Class<?> viocls) {
-      this.st = st;
-      this.viocls = viocls;
+    
+    /**
+     * statement is the statement in the {@link ExecutableSequence} where the failure occurred.
+     */
+    public final Statement statement;
+    
+    /**
+     * violationClass is the type of the {@link ObjectContract}
+     */
+    public final Class<?> violationClass;
+    
+    public Failure(Statement st, Class<?> viocls) {
+      this.statement = st;
+      this.violationClass = viocls;
     }
+    
     public boolean equals(Object o) {
-      if (o == null) return false;
-      if (o == this) return true;
-      Failure other = (Failure)o;
-      if (!st.equals(other.st)) return false;
-      if (!viocls.equals(other.viocls)) return false;
-      return true;
+      if (o instanceof Failure) {    
```

Comment:
It's more idiomatic to test `if (! o instanceof Failure)` and immediately return false.  That uses a common coding pattern, and it keeps from having long indented blocks that the reader must keep track of.  I suggest using the standard idiom throughout (even though I agree yours is equally correct!).